src/test/java/com/uid2/shared/store/RotatingServiceLinkStoreTest.java

```diff
@@ -77,20 +80,45 @@ public void loadContentMultipleServices() throws Exception {
     @Test
     public void findServiceLinksMultipleServices() throws Exception {
         JsonArray content = new JsonArray();
-        ServiceLink l1 = addServiceLink(content, "abc123", 1, 123, "Test Service 1");
-        ServiceLink l2 = addServiceLink(content, "abc123", 2, 123, "test1");
-        ServiceLink l3 = addServiceLink(content, "ghi789", 1, 123, "Test Service 1");
-        ServiceLink l4 = addServiceLink(content, "jkl1011", 3, 124, "test2");
+        ServiceLink l1 = addServiceLink(content, "abc123", 1, 123, "Test Service 1", Set.of());
+        ServiceLink l2 = addServiceLink(content, "abc123", 2, 123, "test1", Set.of(Role.MAPPER));
+        ServiceLink l3 = addServiceLink(content, "ghi789", 1, 123, "Test Service 1", Set.of(Role.MAPPER, Role.SHARER));
+        ServiceLink l4 = addServiceLink(content, "jkl1011", 3, 124, "test2", null);
         when(cloudStorage.download("locationPath")).thenReturn(makeInputStream(content));
 
         final long count = serviceLinkStore.loadContent(makeMetadata("locationPath"));
 
-        ServiceLink sl = serviceLinkStore.getServiceLink(1, "abc123");
-        assertNotNull(sl);
-        assertEquals("Test Service 1", sl.getName());
-        assertEquals(1, sl.getServiceId());
-        assertEquals(123, sl.getSiteId());
-        assertEquals("abc123", sl.getLinkId());
+        ServiceLink sl1 = serviceLinkStore.getServiceLink(1, "abc123");
```

Comment:
The below assertions can be improved by simplifying the equality checks.
* You currently have the four service links l1, l2, l3, l4. You can put all of these into a list `expected`
* Do `serviceLinkStore.getServiceLink(..., ...)` for each entry, and put them into a list `actual`
* Directly compare the equality of the `expected` and `actual` list. This also ensures your implementation of `ServiceLink.equals()` is correct